libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationViewContext.kt

```diff
@@ -72,6 +73,8 @@ internal class DropInNavigationViewContext(
         BitmapMemoryCache(4 * MB_IN_BYTES)
     )
 
+    fun geocoder() = Geocoder(accessToken)
```

Comment:
We can initialize the Geocoder directly inside `GeocodingComponent#onAttached()` as access token is accessible via `mapboxNavigation.navigationOptions`

Reply:
Instantiating inside `GeocodingComponent` will tightly couple both classes , violate [Dependency inversion principle](https://en.wikipedia.org/wiki/Dependency_inversion_principle) and will make it impossible to inject mock instances in tests.

AFAIK `DropInNavigationViewContext` is our IOC container. This setup centralizes creation of `Geocoder` instances, helps us control their scope and opens for us a possibility for `Geocoder` injection from the host application.

Reply:
Yeah these are fair points. I like how we are handling accessToken in `AudioGuidanceViewModel` and `AudioGuidanceServices` without the need to introduce accessToken in `DropInNavigationViewContext`. Do you think if we could do something similar? Also, it's fine if we implement this feature such that `GeocodingComponent` only works with `Geocoder` because there is no plan to allow users to specify a different `Geocoder`